saleor/dashboard/menu/views.py

```diff
@@ -25,8 +26,23 @@ def menu_list(request):
     menus = get_paginator_items(
         menu_filter.qs, settings.DASHBOARD_PAGINATE_BY,
         request.GET.get('page'))
-    ctx = {'menus': menus, 'filter_set': menu_filter,
-           'is_empty': not menu_filter.queryset.exists()}
+    site = get_current_site(request)
+    site_settings = site.settings
+    assign_menu_form = AssignMenuForm(
+        request.POST or None, instance=site_settings, user=request.user)
+    if all([
+            request.method == 'POST',
+            assign_menu_form.is_valid(),
+            request.user.has_perm('menu.edit_menu')]):
```

Comment:
I don't think a form should necessarily know about permissions, that's usually controller-level logic. I would however check the permission before passing data to the form as it's possible to accidentally expose information otherwise (for example by showing the error messages to an unprivileged user or by putting side effects in the clean method).